tests/e2e-tests/src/resolver/dnssec/regression.rs

```diff
@@ -0,0 +1,62 @@
+use std::net::Ipv4Addr;
+
+use dns_test::{
+    client::{Client, DigSettings},
+    name_server::{Graph, NameServer, Sign, SignSettings},
+    record::{Record, RecordType},
+    Implementation, Network, Resolver, Result, FQDN,
+};
+
+#[test]
+#[ignore]
+pub fn gh2252_infinite_recursion_with_deprecated_algorithm() -> Result<()> {
```

Comment:
Instead of the somewhat cryptic `gh2252_` prefix, can we have a docstring that includes a link to the issue?